File: array/229.majority_element_ii.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const majorityElement = nums => {
    const mp = {};

    for (let i = 0; i < nums.length; i++) {
        if (!mp[nums[i]]) mp[nums[i]] = 1;
        else mp[nums[i]]++;
    }

    const keys = Object.keys(mp);
    const r = nums.length / 3;
    const ans = [];
    keys.map(key => {
        if (mp[key] > r) ans.push(key);
    });

    return ans;
};

File: leetcode/array/229_majority_element_ii.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    vector<int> majorityElement(vector<int> &a)
    {
        map<int, int> mp;
        int len = a.size();
        for (int i = 0; i < len; i++)
        {
            mp[a[i]]++;
        }

        int r = len / 3;

        vector<int> ans;

        for (auto x : mp)
        {
            if (r < x.second)
            {
                ans.push_back(x.first);
            }
        }
        return ans;
    }
};

int main()
{

    return 0;
};
// github.com/p-nerd (Shihab Mahamud)
// Saturday, November 19, 2022 | 09:13:25 PM (+06)
